### PR TITLE
Add code samples for invokeMethod

### PR DESCRIPTION
**Pull request by mravn-google**
Addresses https://github.com/flutter/flutter/issues/16112

### CODE REVIEW

File: packages/flutter/lib/src/services/platform_channel.dart [hunk 1]
```diff
@@ -135,6 +135,94 @@ class MethodChannel {
   /// * a [MissingPluginException], if the method has not been implemented by a
   ///   platform plugin.
   ///
+  /// ## Sample code
```
Comment (goderbauer) [inline packages/flutter/lib/src/services/platform_channel.dart]: +1 to these examples!

File: packages/flutter/lib/src/services/platform_channel.dart [hunk 2]
```diff
@@ -135,6 +135,94 @@ class MethodChannel {
...
+  ///
+  ///   static Future<bool> isLicensed() async {
+  ///     // invokeMethod returns a Future<dynamic>, and we cannot pass that for
+  ///     // a Future<bool>, hence the indirection.
+  ///     final bool result = await _channel.invokeMethod('isLicensed');
+  ///     return result;
+  ///   }
+  ///
+  ///   static Future<List<Song>> songs() async {
+  ///     // invokeMethod deserializes compound invocation results using
```
Comment (goderbauer) [inline packages/flutter/lib/src/services/platform_channel.dart]: I don't understand what this comment is supposed to tell me.

File: packages/flutter/lib/src/services/platform_channel.dart [hunk 3]
```diff
@@ -135,6 +135,94 @@ class MethodChannel {
...
+  ///     // List<dynamic> and Map<dynamic, dynamic>. Post-processing code thus
+  ///     // cannot assume e.g. List<Map<String, dynamic>> even though the
+  ///     // simple values involved would support such a typed container.
+  ///     final List<dynamic> songs = await _channel.invokeMethod('getSongs');
+  ///     return songs.map(Song.fromJson).toList();
+  ///   }
+  ///
+  ///   static Future<void> play(Song song, double volume) async {
+  ///     // Errors occurring on the platform side may be relayed to Dart
```
Comment (goderbauer) [inline packages/flutter/lib/src/services/platform_channel.dart]: I would not expose implementation details ("specially marked results") in the comment here. Can we just say: If x happens on the platform side, invokeMethod throws a PlatformException? I think that's easier to understand.

**Comment by Hixie** [comment]
This is great.

**Comment by mravn-google** [comment]
@goderbauer Thanks for the review. I've made some adjustments and added an iOS snippet.

PTAL

**Review by goderbauer — APPROVED** [review]
LGTM

**Pull request merged**